File: src/main/java/com/xl/can/entity/Tenant.java
package com.xl.can.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("tenant")
public class Tenant implements Serializable {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("tenant_code")
    private String tenantCode;

    @TableField("tenant_name")
    private String tenantName;

    @TableField("logo_url")
    private String logoUrl;

    private Integer status;

    @TableField("contact_name")
    private String contactName;

    @TableField("contact_phone")
    private String contactPhone;

    @TableField("invite_code")
    private String inviteCode;

    @TableField("api_quota")
    private Integer apiQuota;

    @TableField("api_used")
    private Integer apiUsed;

    @TableField("expire_time")
    private LocalDateTime expireTime;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    @TableField("deleted_at")
    private Integer deletedAt;
}

File: frontend/src/types/tenant.ts
export interface TenantItem {
  id: number
  tenantCode: string
  tenantName: string
  logoUrl?: string
  contactName?: string
  contactPhone?: string
  status: number
  inviteCode?: string
  apiQuota?: number
  apiUsed?: number
  expireTime?: string
  shopCount: number
  userCount: number
  createdAt: string
  updatedAt?: string
}

export interface TenantDetail {
  id: number
  tenantCode: string
  tenantName: string
  logoUrl?: string
  status: number
  contactName?: string
  contactPhone?: string
  inviteCode?: string
  apiQuota?: number
  apiUsed?: number
  expireTime?: string
  createdAt: string
  updatedAt?: string
}

export interface TenantCreateForm {
  tenantName: string
  logoUrl?: string
  contactName?: string
  contactPhone?: string
  apiQuota?: number
  expireTime?: string
}

export interface TenantEditForm {
  tenantName: string
  logoUrl?: string
  contactName?: string
  contactPhone?: string
  apiQuota?: number
  expireTime?: string
  status: number
}

export interface TenantCreateResult {
  id: number
  tenantCode: string
  tenantName: string
  inviteCode: string
  status: number
}
